src/internal/agent/hooks/argocd-application.go

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package hooks contains the mutation hooks for the Zarf agent.
+package hooks
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/defenseunicorns/zarf/src/config/lang"
+	"github.com/defenseunicorns/zarf/src/internal/agent/operations"
+	"github.com/defenseunicorns/zarf/src/internal/agent/state"
+	"github.com/defenseunicorns/zarf/src/pkg/message"
+	"github.com/defenseunicorns/zarf/src/pkg/transform"
+	"github.com/defenseunicorns/zarf/src/pkg/utils"
+	"github.com/defenseunicorns/zarf/src/types"
+	v1 "k8s.io/api/admission/v1"
+)
+
+type Source struct {
+	Path           string `json:"path"`
+	RepoURL        string `json:"repoURL,omitempty"`
+	TargetRevision string `json:"targetRevision,omitempty"`
+}
+
+type ArgoApplication struct {
+	Spec struct {
+		Source Source `json:"source"`
```

Comment:
In addition to `Source` we may also want to edit `Sources` https://github.com/argoproj/argo-cd/blob/master/pkg/apis/application/v1alpha1/types.go#L84